### PR TITLE
bpo-24391: Better reprs for threading objects.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Add reprs for Semaphore, BoundedSemaphore, Event, and Barrier.

https://bugs.python.org/issue24391

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2020-05-30-10-48-04.bpo-24391.ZCTnhX.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improved reprs of :mod:`threading` synchronization objects.
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Library/2020-05-30-10-48-04.bpo-24391.ZCTnhX.rst]: I think the relevant classes should be listed in the news entry.

File: Lib/threading.py [hunk 2]
```diff
@@ -401,6 +401,12 @@ def __init__(self, value=1):
         self._cond = Condition(Lock())
         self._value = value
 
+    def __repr__(self):
+        return '<%s.%s: %r at %#x>' % (
```
Comment (ZackerySpytz) [inline Lib/threading.py]: It would be nice to use f-strings.

**Review by vstinner — APPROVED** [review]
LGTM, but I concur with @ZackerySpytz: it would be nice to list enhanced classes in the NEWS entry.

**Review by vstinner — APPROVED** [review]
LGTM, thanks!

**Pull request merged**